package ru.yandex.practicum.service.converter;

import org.apache.avro.specific.SpecificRecordBase;
import org.springframework.stereotype.Component;
import ru.yandex.practicum.grpc.telemetry.event.SensorEventProto;
import ru.yandex.practicum.grpc.telemetry.event.SwitchSensorProto;
import ru.yandex.practicum.kafka.telemetry.event.SwitchSensorAvro;

@Component
public class SwitchSensorConverter implements SensorEventConverter {

    @Override
    public SpecificRecordBase convert(SensorEventProto event) {
        SwitchSensorProto switchSensor = event.getSwitchSensor();
        return SwitchSensorAvro.newBuilder()
                .setStat(switchSensor.getState())
                .build();
    }

    @Override
    public SensorEventProto.PayloadCase getType() {
        return SensorEventProto.PayloadCase.SWITCH_SENSOR;
    }
}